apps/veyfi/components/ClaimTab.tsx

```diff
@@ -0,0 +1,74 @@
+import {useVotingEscrow} from '@veYFI/contexts/useVotingEscrow';
+import {useTransaction} from '@veYFI/hooks/useTransaction';
+import * as VotingEscrowActions from '@veYFI/utils/actions/votingEscrow';
+import {validateNetwork} from '@veYFI/utils/validations';
+import {Button} from '@yearn-finance/web-lib/components/Button';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {formatBN, formatUnits} from '@yearn-finance/web-lib/utils/format.bigNumber';
+import {getTimeUntil} from '@yearn-finance/web-lib/utils/time';
+import {useWallet} from '@common/contexts/useWallet';
+
+import {AmountInput} from '../../common/components/AmountInput';
+
+import type {ethers} from 'ethers';
+import type {ReactElement} from 'react';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+
+function ClaimTab(): ReactElement {
+	const {provider, address, isActive, chainID} = useWeb3();
+	const {refresh: refreshBalances} = useWallet();
+	const {votingEscrow, positions, refresh: refreshVotingEscrow} = useVotingEscrow();
+	const refreshData = (): unknown => Promise.all([refreshVotingEscrow(), refreshBalances()]);
+	const [withdrawUnlocked, withdrawUnlockedStatus] = useTransaction(VotingEscrowActions.withdrawUnlocked, refreshData);
+
+	const web3Provider = provider as ethers.providers.Web3Provider;
+	const userAddress = address as TAddress;
```

Comment:
Instead of casting here, can we have a type guard when the address is being used? Something like:
```typescript
function isTAddress(address?: string | null): address is TAddress {
	const regex = /^0x([0-9a-f][0-9a-f])*$/i;
	return !!address && regex.test(address);
}
```
and then
```typescript
const executeWithdrawUnlocked = (): void => {
	if (!votingEscrow  || !address) {
		return;
	}
	if (isTAddress(address)) {
		withdrawUnlocked(web3Provider, address, votingEscrow.address);
	}
};
```

Otherwise we might be casting a non valid address and giving it to `withdrawUnlocked`

Reply:
@Majorfi, if we don't have it yet, I think that function could even live in the web-lib

Reply:
Created a PR https://github.com/yearn/web-lib/pull/160

Reply:
Yes, that's actually even better, we can reuse the `isTAddress()` in the `useWeb3()` hook so that when you destructure the `address` is already of type TAddress. PR https://github.com/yearn/web-lib/pull/162